src/components/Book.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+
+function BookList() {
+  const [books, setBooks] = useState([]);
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const newBook = { title, author };
+    setBooks([...books, newBook]);
+    setTitle('');
+    setAuthor('');
+  };
+
+  return (
+    <div>
+      <h2>Book List</h2>
+      <ul>
+        {books.map((book) => (
+          <li key={null}>
+            {book.title}
+            by
+            {book.author}
+          </li>
```

Comment:
Well done 💯 you're almost there but this part should be a component of its own, I'd suggest renaming this file to `BookList.js` and creating a new `Book.js` component that would have the content of this one, and then you'd import it in this file, it would be a bit like below then 👇🏽 

```
 {books.map((book) => (
    <Book key={null} title={book.title} author={book.author} />
 ))}
```
Make sure you understand [how to pass props from parent (BookList) to children (Book)](https://www.altogic.com/blog/passing-data-from-parent-to-child-in-react#:~:text=Props%2C%20short%20for%20properties%2C%20are,component%20to%20a%20child%20component.&text=In%20this%20example%2C%20the%20parent,object%20and%20renders%20the%20value.) 🔍